server/core/src/logic/user_operations.rs

```diff
@@ -1,56 +1,70 @@
-use self::errors::{ActivateAccountError, AuthorizeUserError, SaveUserError};
+use self::errors::{ActivateAccountError, AuthorizeUserError, SaveUserError, VerifyLoginError};
 
 use crate::config;
-use crate::dto::users::NewUserRequest;
+use crate::dto::users::{LoginRequest, NewUserRequest, UserResponse};
 use crate::email::{render_template, send_mail, ACTIVATION_EMAIL_TEMPLATE};
 use chrono::Duration;
 use entity::account_activation::{self, Entity as AccountActivation};
 use entity::user::{self, Entity as User};
 use entity::Id;
 use lettre::AsyncTransport;
-use migration::DbErr;
 use pwhash::bcrypt;
 use sea_orm::prelude::Uuid;
 use sea_orm::ActiveValue::NotSet;
 use sea_orm::{
-    ActiveModelTrait, ColumnTrait, DatabaseConnection, EntityTrait, IntoActiveModel, QueryFilter,
-    Set, TransactionTrait,
+    ActiveModelTrait, ColumnTrait, DatabaseConnection, DbErr, EntityTrait, IntoActiveModel,
+    QueryFilter, Set, TransactionTrait,
 };
 use std::collections::HashMap;
 use std::sync::Arc;
 
-pub async fn find_user_by_email(
+use super::common::find_entity_by_id;
+
+pub async fn find_user_by_id(
     conn: &DatabaseConnection,
-    email: String,
-) -> Result<Option<user::Model>, DbErr> {
-    let user = User::find()
-        .filter(user::Column::Email.eq(email.clone()))
-        .one(conn)
-        .await?;
+    user_id: Id,
+) -> Result<Option<UserResponse>, DbErr> {
+    let user = find_entity_by_id::<user::Entity>(conn, user_id)
+        .await?
+        .map(|user| user.into());
     Ok(user)
 }
 
-pub async fn find_user_by_id(
+pub async fn verify_login(
     conn: &DatabaseConnection,
-    id: Id,
-) -> Result<Option<user::Model>, DbErr> {
-    let user = User::find_by_id(id).one(conn).await?;
-    Ok(user)
+    req: LoginRequest,
+) -> Result<Id, VerifyLoginError> {
```

Comment:
I feel like this _MethodName+Error_ error name pattern is our next Type postfix like nemesis.

All the code around this would be more readable with _LoginError_

Reply:
I am not sure about this, mainly because I do not know what to name some errors to not be like this but I see your point. I am bad at naming 😓 . Can you rename our errors to be more readable? Alternatively, if we do another PP we can do it there quickly. I will open an issue and it can be done at a later date.